fdbclient/include/fdbclient/FDBTypes.h

```diff
@@ -750,11 +750,38 @@ struct GetRangeLimits {
 
 struct RangeResultRef : VectorRef<KeyValueRef> {
 	constexpr static FileIdentifier file_identifier = 3985192;
-	bool more; // True if (but not necessarily only if) values remain in the *key* range requested (possibly beyond the
-	           // limits requested) False implies that no such values remain
-	Optional<KeyRef> readThrough; // Only present when 'more' is true. When present, this value represent the end (or
-	                              // beginning if reverse) of the range which was read to produce these results. This is
-	                              // guaranteed to be less than the requested range.
+	bool more; // True if values remain in the *key* range requested (possibly beyond the
+	           // limits requested), but not necessarily only if, for example, in getRangeStream(), 'more' is always set
+	           // to true, as each stream fragment doesn't know if it's the last one. Instead, it uses
+	           // `error_code_end_of_stream` for the end.
+	           // False implies that no such values remain
+
+	// Only present when 'more' is true, for example, when the read reaches the shard boundary, 'readThrough' is set to
+	// the shard boundary and the client's next range read should start with the 'readThrough'.
+	// But 'more' is true does not necessarily guarantee 'readThrough' is present, for example, when the read reaches
+	// size limit, 'readThrough' might not be set, the next read should just start from the keyAfter of the current
+	// query result's last key.
+	// In both cases, please use the getter function 'getReadThrough()' instead, which represents the end (or beginning
+	// if reverse) of the range which was read.
```

Comment:
You could enforce this more strongly by making `readThrough` private

Reply:
There're a few places want to access `readThrough` directly, e.g. `IKVSReadRangeReply` to serialize the `RangeResult`, and maybe `MappedRangeResultRef`. Also a few ASSERT() which might be useful to keep.